### ISSUE TITLE
White light from RGB and RGBW

### ISSUE THREAD

**Issue by Weldon26** [dotsam/homebridge-milight#5]
My rgb strips and rgbw bulbs are showing up fine in all my HomeKit apps. Awesome! But I can NOT figure out what I've done wrong in my config or what doc I didn't read that explains why I can't set their colors to white. 
My rgb strips do a pretty good job of pretending they're white LED's. But There's no white in the color pickers for those strips in any of the HomeKit apps I've tried. My bulbs produce a great white, but, again, I can't figure out how to set them to white mode in any of my apps. 
Is this due to a mistake in my config? Or am I using the apps wrong when setting my milights? Or is white not currently supported? I'm thrilled with what the plugin is already doing for me. I just want to make sure I'm using it properly and getting the most from it.

**Comment by dotsam** [comment]
I'm actually working on this exact issue as we speak. The MiLight bulbs don't support saturation, only brightness and hue, so I hadn't previously been including that characteristic for the bulbs. But it looks like depending on the implementation of the app, this prevents the colour from being set to white.

In the HSV parlance that HomeKit uses, white should be Hue = 0, Saturation = 0, Brightness = 100. Without saturation being able to be set to 0, you can never get white, so some apps won't display it at all. I had been mainly using the Insteon app, which has a colour wheel, with one of the options as white, which sets the hue to 0 regardless of saturation being an option. Other apps though, like Eve, won't present the right controls to get to white without the saturation characteristic being present.

Long story short, it's app-dependant right now, but with an upcoming release, it should be working in almost any app.

**Comment by Weldon26** [comment]
Thanks for the info! Glad to know I may not be quite as dumb as I was feeling there for a while. I was sure _I_ was the problem -- not the plug-in or the apps. And thanks for reminding me of the Insteon app. It was my favorite HomeKit app early on and somehow I had forgotten all about it. I'll stick to using it for now. Problem solved. :)

**Comment by dotsam** [comment]
This should all be fixed in 0.1.0 which I just pushed.
